pkg/reconciler/labeler/labels.go

```diff
@@ -90,38 +89,22 @@ func (c *Reconciler) clearLabels(ctx context.Context, ns, name string) error {
 // setLabelForListed uses the accessor to attach the label for this route to every element
 // listed within "names" in the same namespace.
 func setLabelForListed(ctx context.Context, route *v1alpha1.Route, acc accessor, names sets.String) error {
-	nameToAccessor := make(map[string]kmeta.Accessor)
-
-	// Lookup names that are missing our Route label.
 	for name := range names {
 		elt, err := acc.get(route.Namespace, name)
 		if err != nil {
 			return err
 		}
-		nameToAccessor[name] = elt
 		routeName, ok := elt.GetLabels()[serving.RouteLabelKey]
-		if !ok {
-			continue
-		}
-		if routeName != route.Name {
-			return fmt.Errorf("%s %q is already in use by %q, and cannot be used by %q",
-				elt.GroupVersionKind(), elt.GetName(), routeName, route.Name)
-		}
-	}
-
-	// Set label for newly added names as traffic target.
-	for _, name := range names.List() {
-		elt := nameToAccessor[name]
-		if elt.GetLabels() == nil {
-			elt.SetLabels(make(map[string]string))
-		} else if _, ok := elt.GetLabels()[serving.RouteLabelKey]; ok {
-			continue
-		}
-
-		if err := setRouteLabel(acc, elt, &route.Name); err != nil {
-			logging.FromContext(ctx).Errorf("Failed to add route label to %s %q: %s",
-				elt.GroupVersionKind(), elt.GetName(), err)
-			return err
+		if ok {
+			if routeName != route.Name {
+				return fmt.Errorf("%s %q is already in use by %q, and cannot be used by %q",
+					elt.GroupVersionKind(), elt.GetName(), routeName, route.Name)
```

Comment:
this naming feels prone to errors :-) I had to read it twice before I saw `.` :)